arkouda/client.py

```diff
@@ -539,14 +537,14 @@ def generic_msg(cmd : str, args : Union[str,bytes]=None, send_bytes : bool=False
         raise RuntimeError("client is not connected to a server")
     
     try:
-        if send_bytes:
-            return _send_binary_message(cmd=cmd, 
-                                            payload=cast(bytes,args), 
-                                            recv_bytes=recv_bytes)         
+        if send_binary:
+            assert payload is not None
+            return _send_binary_message(cmd=cmd, payload=payload,
+                                        recv_binary=recv_binary, args=args)
         else:
-            return _send_string_message(cmd=cmd, args=cast(str,args), 
-                                            recv_bytes=recv_bytes)
-                
+            assert payload is None
+            return _send_string_message(cmd=cmd, args=args, recv_binary=recv_binary)
+
```

Comment:
Minor: instead of the assertions, the payload conditions could be moved into the if/else conditional logic and a third case could be added to handle the assertion failures with warning messages.
i.e.
```
if send_binary and payload is not None:
  return _send_binary....
elif not send_binary and payload is None:
  return _send_string...
else:
  warnings.warn(..you forgot something...)
```